core/src/test/resources/sql/planner.iq

```diff
@@ -132,12 +132,13 @@ EnumerableCalc(expr#0..2=[{inputs}], $f0=[$t1], $f1=[$t2])
 
 # [CALCITE-7000] Extend IntersectToSemiJoinRule to support n-way inputs
 !set planner-rules "-EnumerableRules.ENUMERABLE_INTERSECT_RULE,
+  -CoreRules.INTERSECT_TO_DISTINCT,
   +CoreRules.INTERSECT_TO_SEMI_JOIN"
 select a from (values (1), (2), (3), (4), (5)) as t1 (a)
 intersect
 select a from (values (1), (2)) as t2 (a)
 intersect
-select a from (values (1), (4)) as t3 (a);
+select a from (values (1), (4), (null)) as t3 (a);
```

Comment:
I think the idea is to run the same test with both old and new rules and see that the results are the same
(there is no way to say "the same", but the table output should be the same).
So can you also run it once with the existing rules?

Reply:
Thank you for your detailed review. The reason for this result is that there is no good way to control the order of rules, as you mentioned before. This is the only way to do it now, and it may be the same for JdbcTest. I will think about what good way to control the order later.

Reply:
I don't think the order is important. Can you run the same query before you modify the rules?

Reply:
Sorry, I didn't explain it clearly. It is because the modified rule depends on INTERSECT_MERGE, and if we had done INTERSECT_TO_SSI_JOIN first, it would have produced the current result.

Reply:
is there a way to run this example in quidem with and without the new rule and compare results?

Reply:
I have currently modified the code and it seems that I can no longer execute the rules before the modification. Do you want to run the 2-way rule? If you want to run without INTERSECT_TO_SEMI_JOIN, there is currently a case in the submission that uses  INTERSECT for calculations.

Reply:
I don't really understand what you are saying.
What does it mean "I can no longer execute the rules before the modification"?
You are not executing any rules, the compiler is, when preparing a test.